homeassistant/components/zimi/light.py

```diff
@@ -0,0 +1,106 @@
+"""Light platform for zcc integration."""
+
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from zcc import ControlPoint
+from zcc.device import ControlPointDevice
+
+from homeassistant.components.light import ATTR_BRIGHTNESS, ColorMode, LightEntity
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddConfigEntryEntitiesCallback
+
+from . import ZimiConfigEntry
+from .entity import ZimiEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ZimiConfigEntry,
+    async_add_entities: AddConfigEntryEntitiesCallback,
+) -> None:
+    """Set up the Zimi Light platform."""
+
+    api = config_entry.runtime_data
+
+    lights: list[ZimiLight | ZimiDimmer] = [
+        ZimiLight(device, api)
+        for device in api.lights if device.type != "dimmer"
+    ]
+
+    lights.extend(
+        [
+            ZimiDimmer(device, api)
+            for device in filter(lambda light: light.type == "dimmer", api.lights)
```

Comment:
My comment about replacing filter + lambda with a comprehension was a general comment.
```suggestion
            for device in api.lights if device.type == "dimmer"
```

Reply:
Sorry, I realised that a few moments ago when I merged into my stage 2 branch and noticed the next one!